### DISCUSSION TITLE
How should I choose between @volar/language-service and @volar/kit?

### DISCUSSION THREAD

**Discussion by alamhubb** [volarjs/volar.js#249] [Q&A]
Hello, I am developing an LSP language plug-in

@volar/kit is a package of @volar/language-service. Does it mean that @volar/kit should be used first?

But I see that the official examples do not use @volar/kit first, but @volar/language-service. Why is that?

If not, in what cases should @volar/kit be used and in what cases should @volar/language-service be used?

Or should @volar/language-service be used first, and only when @volar/language-service cannot meet the needs, then try to use @volar/kit?

@volar/language-service
This module provides language service functionalities, such as offering IntelliSense features. It depends on @volar/language-core for obtaining and processing virtual code, and then provides corresponding language services.

@volar/kit
@volar/kit is a module that encapsulates @volar/language-service. It provides a simplified interface for using Volar's diagnostic and formatting features within Node.js applications.

**Comment by alamhubb** [comment]
My current understanding is that kit should not be used when developing LSP. Kit is suitable for non-LSP scenarios. I don't know if my understanding is correct.

**Reply by remcohaszing** [reply]
I’m not exactly sure what `@volar/kit` is. For https://github.com/mdx-js/mdx-analyzer I don’t use `@volar/kit` (yet), so you should be able to get started without it.

**Reply by alamhubb** [reply]
> My current understanding is that kit should not be used when developing LSP. Kit is suitable for non-LSP scenarios. I don't know if my understanding is correct.

Thanks for your answers
